### PR TITLE
fix: fill missing price-history graph datapoints with 0

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Fixes https://app.clickup.com/t/9002019994/PRF-7046

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-7046 On cysync in portfolio 1 year graph duration isn't correct](https://app.clickup.com/t/9002019994/PRF-7046)

File: packages/coin-support-utils/src/getAccountHistory/index.ts [hunk 1]
```diff
@@ -52,6 +52,29 @@ async function getTransactions(
   );
 }
 
+function fillMissingPriceHistory(
+  history: IPriceSnapshot[],
+  days: 1 | 7 | 30 | 365,
+) {
+  const expectedEntries = days === 365 ? days : days * 24;
+  const startTime = history[0].timestamp;
+  const interval = days === 365 ? 24 * 60 * 60 * 1000 : 60 * 60 * 1000;
```
Comment (irshadCypherock) [inline packages/coin-support-utils/src/getAccountHistory/index.ts]: This is also defined in line `123`. It's better if we have only 1 definition

File: packages/coin-support-utils/src/getAccountHistory/index.ts [hunk 2]
```diff
@@ -52,6 +52,29 @@ async function getTransactions(
   );
 }
 
+function fillMissingPriceHistory(
+  history: IPriceSnapshot[],
+  days: 1 | 7 | 30 | 365,
+) {
+  const expectedEntries = days === 365 ? days : days * 24;
```
Comment (irshadCypherock) [inline packages/coin-support-utils/src/getAccountHistory/index.ts]: Same for this

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/coin-support-utils/src/getAccountHistory/index.ts (hunk 1 above)
Comment (ujjwal-cyph) [reply packages/coin-support-utils/src/getAccountHistory/index.ts]: changed

File: packages/coin-support-utils/src/getAccountHistory/index.ts (hunk 2 above)
Comment (ujjwal-cyph) [reply packages/coin-support-utils/src/getAccountHistory/index.ts]: changed

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**